docs/features/async-streams.md

```diff
@@ -14,8 +14,8 @@ namespace System.Collections.Generic
 
     public interface IAsyncEnumerator<out T> : System.IAsyncDisposable
     {
-        System.Threading.Tasks.ValueTask<bool> WaitForNextAsync();
-        T TryGetNext(out bool success);
+        System.Threading.Tasks.ValueTask<bool> MoveNextAsync();
+        T Current { get; }
```

Comment:
@agocke @cston @dotnet/roslyn-compiler for review.
The main change in this PR is to use simpler ` MoveNextAsync()/Current` pattern for async-streams, instead of `WaitForNextAsync()/TryGetNext()`, per last week's LDM. This much simplifies the code.
Also fixing some minor issues and adding tests for PROTOTYPE comments.

We're aiming to get async-streams merged to dev16.0.x before Friday. I expect to have one or two small PRs to address blocking issues following this change.
Thanks